### PR TITLE
Pootle init: improvements

### PR DESCRIPTION
**Pull request by phlax**
This commit:

- Separates the init_command from the app_runner logic
- Provides correct usage/help for pootle command
- Provides correct usage/help for subcommands

### CODE REVIEW

File: pootle/runner.py [hunk 1]
```diff
@@ -130,78 +200,41 @@ def run_app(project, default_settings_path, settings_template,
         will be set to.
     """
     runner_name = os.path.basename(sys.argv[0])
-    src_dir = os.path.abspath(os.path.dirname(__file__))
-
-    parser = ArgumentParser()
 
+    # This parser should ignore the -h flag, unless there is no subcommand
```
Comment (unho) [inline pootle/runner.py]: `--help` is easier to spot and understand.

File: pootle/runner.py [hunk 2]
```diff
@@ -130,78 +200,41 @@ def run_app(project, default_settings_path, settings_template,
...
+    parser = ArgumentParser(add_help=False)
     parser.add_argument("--config",
                         default=default_settings_path,
                         help=u"Use the specified configuration file.")
     parser.add_argument("--noinput", action="store_true", default=False,
                         help=u"Never prompt for input")
-    parser.add_argument("--version", action="version", version=get_version())
```
Comment (unho) [inline pootle/runner.py]: Is it not better to keep this option?

Comment (phlax) [inline pootle/runner.py]: k - i have and it parses that straight away

**Comment by julen** [comment]
Can we have this split into separate commits please?

**Comment by phlax** [comment]
not easily - the refactoring of `pootle_init` is contingent on other changes in the app runner to work correctly.

**Comment by julen** [comment]
Realize reviewing is not easy this way, and the "not easy" part is left on anyone willing to check this out (both now and in the future).

File: pootle/runner.py (hunk 2 above)
Comment (unho) [inline pootle/runner.py]: ok

**Comment by unho** [comment]
lgtm from my side.

File: pootle/runner.py [hunk 3]
```diff
@@ -86,6 +93,69 @@ def init_settings(settings_filepath, template_filename,
             settings.write(template.read() % context)
 
 
+def init_command(parser, settings_template, args):
+    """ Parse and run the `pootle init` command
```
Comment (julen) [inline pootle/runner.py]: Nit: initial whitespace

File: pootle/runner.py [hunk 4]
```diff
@@ -86,6 +93,69 @@ def init_settings(settings_filepath, template_filename,
...
+    :param parser: `argparse.ArgumentParser` instance to use for parsing
+    :param settings_template: Template file for initializing settings from.
+    :param args: Arguments to call init command with.
+    """
+
+    src_dir = os.path.abspath(os.path.dirname(__file__))
+    add_help_to_parser(parser)
+    parser.add_argument("--db", default="sqlite",
+                        help=(u"Use the specified database backend (default: "
+                              "'sqlite'; other options: 'mysql', "
```
Comment (julen) [inline pootle/runner.py]: I think you mean to use `u''` strings here and elsewhere.

**Comment by julen** [comment]
GTM after nits are fixed.

**Pull request merged**